### PR TITLE
TR-736 added amp_html support for snippets and templates behind flag

### PR DESCRIPTION
**Pull request by ischer**

### CODE REVIEW

File: src/components/contentEditor/ContentEditor.js [hunk 1]
```diff
@@ -52,25 +59,36 @@ class ContentEditor extends React.Component {
   // note, must handle null template parts
```
Comment (ischer) [inline src/components/contentEditor/ContentEditor.js]: i couldn't find a case where these values could be `null`, so i changed the condition below.  is there a way to get a `null` value?

Comment (ischer) [reply src/components/contentEditor/ContentEditor.js]: just tested this out by creating a few templates. since `normalize` defaults the value to `''`, the undefined values are transformed to empty strings during the check. i think this should be okay.

**Comment by ischer** [comment]
just found a small bug that i'll also be fixing :tmyk:
the published template view does not seem to be showing the amp_html tab

File: src/components/contentEditor/ContentEditor.js (hunk 1 above)
Comment (ewandennis) [reply src/components/contentEditor/ContentEditor.js]: Ah JS :) 👍

File: src/actions/snippets.js [hunk 2]
```diff
@@ -9,9 +9,11 @@ export const createSnippet = ({
   name,
   sharedWithSubaccounts = false,
   subaccountId,
-  text
-}) => (
-  sparkpostApiRequest({
+  text,
+  amp_html,
+  isAmpLive
```
[the remark this answers is not in the document]
Comment (ewandennis) [reply src/actions/snippets.js]: My comment was incomprehensible on re-reading. Let me try again:

I suggest moving your `isAmpLive` logic from `actions/snippets.js` into `CreatePage` and `EditPage` to keep the snippets action creators as just declarative API call thunks.

So your `createSnippet()` call in `CreatePage` changes to something like this:

```js
    return createSnippet({
      html,
      id,
      name,
      sharedWithSubaccounts: assignTo === 'shared',
      subaccountId,
      text,
      amp_html: isAmpLive ? amp_html : undefined,
      isAmpLive
    }).then(() => ...);
```

Similar for the `updateSnippet()` call in `EditPage`.

File: src/components/contentEditor/ContentEditor.js [hunk 3]
```diff
@@ -52,25 +59,36 @@ class ContentEditor extends React.Component {
...
-    const { readOnly } = this.props;
+    const { readOnly, isAmpLive, contentOnly } = this.props;
     const { selectedTab } = this.state;
 
     const tabs = fields
       .filter((field) => field.show(this.props))
       .map(({ content }, index) => ({ content, onClick: () => this.handleTab(index) }));
 
+    // Templates require HTML or Text. Snippets require HTML, AMP HTML, or Text.
+    const requiredContentValidator = isAmpLive && contentOnly ? this.requiredHtmlTextOrAmp : this.requiredHtmlOrText;
```
[the remark this answers is not in the document]
Comment (ischer) [reply src/components/contentEditor/ContentEditor.js]: i think i've got this covered in `ContentEditor.test.js` with 
```
  it('should set required content validation correctly', () => {
    expect(wrapper.find('Field').props().validate[0]).toBe(wrapper.instance().requiredHtmlOrText);

    wrapper.setProps({ isAmpLive: true, contentOnly: true });
    expect(wrapper.find('Field').props().validate[0]).toBe(wrapper.instance().requiredHtmlTextOrAmp);
  });
```

File: src/components/contentEditor/ContentEditor.test.js [hunk 4]
```diff
@@ -43,15 +43,34 @@ describe('ContentEditor', () => {
...
+    expect(wrapper.find('Field').props().name).toBe('content.amp_html');
+    expect(wrapper.find('Field').props().syntaxValidation).toBe(false);
+
+    wrapper.instance().handleTab(3);
+    wrapper.update();
+    expect(wrapper.find('Field').props().name).toBe('testData');
     expect(wrapper.find('Field').props().syntaxValidation).toBe(true);
   });
 
+  it('should set required content validation correctly', () => {
```
[the remark this answers is not in the document]
Comment (ischer) [reply src/components/contentEditor/ContentEditor.test.js]: could you point me toward an example of what you mean?

File: src/actions/tests/snippets.test.js [hunk 5]
```diff
@@ -61,6 +61,15 @@ describe('Snippet Actions', () => {
           text: 'Testing...'
         })
       )
+    },
+    {
```
Comment (ewandennis) [inline src/actions/tests/snippets.test.js]: This is redundant since `createSnippet()` no longer uses `isAmpLive`.

File: src/actions/tests/snippets.test.js [hunk 6]
```diff
@@ -145,6 +154,15 @@ describe('Snippet Actions', () => {
...
```
Comment (ewandennis) [inline src/actions/tests/snippets.test.js]: This is redundant since `updateSnippet()` no longer uses `isAmpLive`.

File: src/actions/snippets.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ewandennis) [reply src/actions/snippets.js]: Oh man. This comment made you spread `isAmpLive` far and wide. Sorry about that.

File: src/components/contentEditor/ContentEditor.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ewandennis) [reply src/components/contentEditor/ContentEditor.test.js]: Don't worry about it. The test style in here was already off. Its not on you to fix.
